### PR TITLE
[feature] Reduce Number of Queries in Triggered Mails Script [OSF-8167]

### PR DESCRIPTION
**Pull request by caseyrollins**
#### Purpose
- Make `triggered_mails.py` less query intensive.

#### Changes
- MODM --> Django

#### Side Effects
- Possible that I've made a mistake and the the new queries aren't equivalent to the old ones? Tested against staging DB and got the same results for new and old queries though. See notes below.

#### Ticket
- [OSF-8167](https://openscience.atlassian.net/browse/OSF-8167)

#### Notes/Testing
- **OLD**
```
In [14]: len(conn.queries)
Out[14]: 0

In [15]: inactive_users = User.find((MQ('date_last_login', 'lt', timezone.now() - osf_settings.NO_LOGIN_WAIT_TIME) & MQ('tags__name', 'ne', 'osf4m')) | (MQ('date_last_login', 'lt', timezone.now() - osf_settings.NO_LOGIN_OSF4M_WAIT_TIME) & MQ('tags__name', 'eq', 'osf4m')))

In [16]: inactive_emails = QueuedMail.find(MQ('email_type', 'eq', NO_LOGIN_TYPE))

# 1373 queries for 686 inactive_emails
In [17]: users_sent_id = [email.user._id for email in inactive_emails] 

In [18]: inactive_ids = [user._id for user in inactive_users if user.is_active]

In [19]: users_to_send = [User.load(id) for id in (set(inactive_ids) - set(users_sent_id))]

In [20]: len(conn.queries) 
==========> Out[20]: 1375 <==========

In [26]: len(users_sent_id)
Out[26]: 686

In [27]: len(users_to_send)
Out[27]: 0
```

- **NEW**
```
In [34]: len(conn.queries)
Out[34]: 0

In [35]: new_users_sent_ids = QueuedMail.objects.filter(email_type=NO_LOGIN_TYPE).values_list('user__guids___id')

In [36]: new_users_to_send = User.objects.filter((Q(date_last_login__lt=timezone.now() - osf_settings.NO_LOGIN_WAIT_TIME) & ~Q(tags__name='osf4m')) | Q(date_last_login__lt=timezone.now() - osf_settings.NO_LOGIN_OSF4M_WAIT_TIME, tags__name='osf4m'), is_active=True).exclude(guids___id__in=new_users_sent_ids)

In [37]: new_users_to_send
Out[37]: <GuidMixinQuerySet []>

In [38]: len(conn.queries) 
==========> Out[38]: 1 <========== 

In [39]: len(new_users_sent_ids)
Out[39]: 686
```

### CODE REVIEW

File: scripts/triggered_mails.py [hunk 1]
```diff
@@ -33,19 +32,13 @@ def main(dry_run=True):
...
-    inactive_emails = QueuedMail.find(Q('email_type', 'eq', NO_LOGIN_TYPE))
-
-    #This is done to prevent User query returns comparison to User, as equality fails
-    #on datetime fields due to pymongo rounding. Instead here _id is compared.
-    users_sent_id = [email.user._id for email in inactive_emails]
-    inactive_ids = [user._id for user in inactive_users if user.is_active]
-    users_to_send = [User.load(id) for id in (set(inactive_ids) - set(users_sent_id))]
-    return users_to_send
-
+    users_sent_ids = QueuedMail.objects.filter(email_type=NO_LOGIN_TYPE).values_list('user__guids___id')
```
Comment (sloria) [inline scripts/triggered_mails.py]: Drive-by: Isn't `flat=True` necessary here?

Comment (caseyrollins) [inline scripts/triggered_mails.py]: Having read the "Performance Optimizations" section of https://docs.djangoproject.com/en/1.11/ref/models/querysets/#in, I should probably change the exclude to be: `.exclude(guids___id__in=list(users_sent_ids))`
